www-src/app/replicator/change/change_event_handler.coffee

```diff
@@ -1,28 +1,68 @@
+AndroidCalendarHandler = require "../../lib/android_calendar_handler"
+CozyToAndroidEvent = require "../transformer/cozy_to_android_event"
 log = require('../../lib/persistent_log')
     prefix: "ChangeEventHandler"
     date: true
 
 module.exports = class ChangeEventHandler
 
-    change: (doc) ->
-        log.info "change"
+    constructor: (@calendarSync) ->
+        @androidCalendarHandler = new AndroidCalendarHandler()
+        @cozyToAndroidEvent = new CozyToAndroidEvent()
+        @calendarSync = navigator.calendarsync unless @calendarSync
 
-        if doc._rev.split('-')[0] is 1
-            @create doc
-        else
-            @update doc
+    dispatch: (cozyEvent) ->
+        log.info "dispatch"
 
-    create: (doc) ->
+        @calendarSync.eventBySyncId cozyEvent._id, (err, androidEvents) =>
+            if androidEvents.length > 0
+                androidEvent = androidEvents[0]
+                if cozyEvent._deleted
+                    @delete cozyEvent, androidEvent
+                else
+                    @update cozyEvent, androidEvent
+            else
+                @create cozyEvent unless cozyEvent._deleted
```

Comment:
Could you add a comment about the `unless cozyEvent._deleted` . This organization is cleaner than https://github.com/kosssi/cozy-mobile/blob/refactoring_replication/www-src/app/replicator/replicator_calendars.coffee#L346-L359 ; but less intuitive (3 possibilities (create, update, delete), but 4 cases ....